ClientApp/viking-vault/src/UserDashboard/EditAutomaticPaymentModal.tsx

```diff
@@ -0,0 +1,181 @@
+import React from "react";
+import { constants } from "../Resources/Constants.js";
+import '../AdminDashboard/AttachCardModal.css';
+import { IAutomaticPayment } from "./AutomaticPaymentList.jsx";
+
+const regexCheckIfPositiveFloat = "^[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?$";
+let currentDate = new Date();
+currentDate.setHours(0);
+
+interface IModalProps {
+    open: boolean;
+    onModalClose: any;
+    automaticPayment: IAutomaticPayment;
+}
+
+interface IAutomaticPaymentProperties {
+    amount: string;
+    initialPaymentDate: string;
+}
+
+interface IFormState {
+    [key: string]: string | boolean | IAutomaticPaymentProperties | null;
+    automaticPayment: IAutomaticPaymentProperties;
+    valid: boolean | null;
+    errorLabel: string;
+}
+
+class EditAutomaticPaymentForm extends React.Component<IModalProps, IFormState> {
+    constructor(props: IModalProps) {
+        super(props);
+        this.state = {
+            automaticPayment: {
+                amount: "",
+                initialPaymentDate: "",
+            },
+            valid: null,
+            errorLabel: "",
+        }
+    }
+
+    private handleChange = (inputValue: string, inputName: string) => {
+        this.setState({
+            automaticPayment: {
+                ...this.state.automaticPayment,
+                [inputName]: inputValue
+            }
+        });
+    }
+
+    private amountMustBePositive = () => {
+        if(this.state.automaticPayment.amount.match(regexCheckIfPositiveFloat)) {
+            return  {
+                message: "Ok",
+                class: "alert alert-success"
+            }
+        }
+        return  {
+            message: "The amount must be a positive number!",
+            class: "alert alert-info"
+        }
+    }
+
+    private initialPaymentDateIsAfterCurrentDate = () => {
```

Comment:
Usually we start with the verb for this kind of function names (i.e. isInitialPaymentDateAfterCurrentDate) so that when we need to use it in an if statement, it`s going to look like a sentence: if (isInitialPaymentDateAfterCurrentDate()) then... This way, you make it very clear that we are talking about a method and not a variable.